benchexec/tablegenerator/react-table/src/components/TaskDefinitionViewer.js

```diff
@@ -0,0 +1,114 @@
+ *
+ * BenchExec is a framework for reliable benchmarking.
+ * This file is part of BenchExec.
+ * Copyright (C) Dirk Beyer. All rights reserved.
+ */
+import React from "react";
+import yamlParser from "yaml";
+
+/** Special view for YAML files in the LinkOverlay component. */
+export default class TaskDefinitionViewer extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      splitterTag: "<splitter#9d81y23>",
+      fileTag: "<file#092nt43>",
+      content: this.props.yamlText,
+    };
+  }
+
+  componentDidMount() {
+    this.prepareTextForRendering();
+  }
+
+  componentDidUpdate(prevProps) {
+    if (prevProps.yamlText !== this.props.yamlText) {
+      this.prepareTextForRendering();
+    }
+  }
+
+  /**
+   * Parses the YAML file and encloses all input and property files with a fileTag as well as a splitterTag,
+   * so they can be rendered separately as links. Takes the following assumptions:
+   * input_files is either a string or a list of strings
+   * properties is a list of dicts, each with a "property_file" key
+   */
+  prepareTextForRendering = () => {
+    const yamlObj = yamlParser.parseDocument(this.props.yamlText);
+
+    const inputFiles = yamlObj.get("input_files");
+    if (inputFiles) {
+      if (Array.isArray(inputFiles.items)) {
+        inputFiles.items.forEach(inputFileItem => {
+          inputFileItem.value = this.encloseFileInTags(inputFileItem.value);
+        });
+      } else {
+        yamlObj.set("input_files", this.encloseFileInTags(inputFiles));
+      }
+    }
+
+    const properties = yamlObj.get("properties");
+    if (properties) {
+      if (Array.isArray(properties.items)) {
+        properties.items.forEach(property => {
+          if (Array.isArray(property.items)) {
+            property.items.forEach(propertyItem => {
+              if (propertyItem.key.value === "property_file") {
+                propertyItem.value.value = this.encloseFileInTags(
+                  propertyItem.value.value,
+                );
+              }
+            });
+          }
+        });
+      }
+    }
+
+    this.setState({ content: yamlObj.toString() });
+  };
+
+  encloseFileInTags = fileName => {
+    return (
+      this.state.splitterTag +
+      this.state.fileTag +
+      fileName +
+      this.state.fileTag +
+      this.state.splitterTag
+    );
+  };
+
+  render() {
+    const contentBySplitter = this.state.content.split(this.state.splitterTag);
+    const jsxContent = [];
+
+    contentBySplitter.forEach(contentPart => {
+      let jsxContentPart;
+      // If contentPart is enclosed with file tags (= if contentPart is a file which should be linked)
+      if (
+        contentPart.match(
+          "^" + this.state.fileTag + "(?:.)+" + this.state.fileTag + "$",
+        )
+      ) {
+        contentPart = contentPart.replace(
+          new RegExp(this.state.fileTag, "g"),
+          "",
+        );
+        jsxContentPart = (
+          <span
```

Comment:
Is there a specific reason for not using `<a>` tags?

Reply:
There was during the development but it doesn't make a difference at this point. `<a>` tags are more suitable here though, so I will change it.

Reply:
I just noticed that this actually does bring a problem with it.
ESLint shows the following warning:

> The href attribute requires a valid value to be accessible. Provide a valid, navigable address as the href value. If you cannot provide a valid href, but still need the element to resemble a link, use a button and change it with appropriate styles. Learn more: https://github.com/evcohen/eslint-plugin-jsx-a11y/blob/master/docs/rules/anchor-is-valid.md  jsx-a11y/anchor-is-valid

 In this case a `href` attribute is not what we want though. Therefore I would suggest keeping the `<span>` tags.

Reply:
I see. But this raises an interesting other question: Is the current solution accessible, i.e., is it usable with a keyboard (using <kbd>Tab</kbd> to focus the "link" and <kbd>Enter</kbd> to activate it)? If no, we should fix that. At least we should probably provide an ARIA role.

Reply:
No, the current solution is not accessible via the keyboard.
How about simply using `<button>` tags for the links (and also the "Back to task definition" link)? The button style can easily be overriden by a few css attributes. Of course we could also add ARIA markup, but if we can use a native html element that already works I wouldn't see an argument against using it.

Reply:
Yes, that seems to make sense.

Reply:
The links could easily be changed to `<button>` tags, however there were some problems with the style of the "Back to task definition" button due to the standard vertical alignment of the `<button>` tag. It seems that the only way to affect the vertical alignment of buttons is with hacky workarounds, so I instead made it behave like a button with some markup.

Also the focus is changed to the body element of the page when new content is loaded into the overlay, so Tab had to be pressed two times to access the button. To avoid this, I added some code that focuses the modal container when updating.

Any way, all links are now accessible via keyboard.

Reply:
Thanks! Does it maybe work if setting `display: inline`? But if not, we can just keep it as it is.

Reply:
No, unfortunately altering the `display` value doesn't work.